### PR TITLE
[runtime] Fix warnings

### PR DESCRIPTION
**Pull request by mbouaziz**
In #135, @jberdine found that we used pointer arithmetic on `void`. This could have been avoided by enabling compiler warnings. So let's do it and fix a few more oddities.

I've run
```shell
set -a; SKARGO_PROFILE=dev; STAGE=2; PATH=$PWD/compiler/stage$STAGE/bin:$PATH; set +a; make clean && make -C compiler clean test && make test
```
as suggested by @jberdine

### CODE REVIEW

File: compiler/runtime/obstack.c [hunk 1]
```diff
@@ -79,14 +79,15 @@ void sk_obstack_attach_page(char* lpage) {
 }
```
Comment (jberdine) [inline compiler/runtime/obstack.c:79]: I can't comment on the commit summary, but does "My VSCode is configured with format-on-save and uses the standard Microsoft C/C++ Extension Pack." mean that it runs clang-format, or does its own thing?

File: compiler/runtime/runtime.c [hunk 2]
```diff
@@ -17,7 +17,6 @@ SKIP_gc_type_t* get_gc_type(char* skip_object) {
   return *slot1;
 }
 
-
```
Comment (jberdine) [inline compiler/runtime/runtime.c:20]: Aren't the *local_context* functions below also dead? And then also the comment right below here.

File: compiler/runtime/runtime.c [hunk 3]
```diff
@@ -83,32 +82,6 @@ void* SKIP_llvm_memcpy(char* dest, char* val, SkipInt len) {
 /* Global context synchronization. */
 /*****************************************************************************/
 
-#ifdef SKIP32
```
Comment (jberdine) [inline compiler/runtime/runtime.c:86]: The comment above goes with this removed code.

File: compiler/runtime/runtime32_specific.c [hunk 4]
```diff
@@ -93,7 +93,7 @@ void sk_context_set_unsafe(char* obj) {
   context = obj;
 }
 
-SkipInt SKIP_genSym(SkipInt n) {
+SkipInt SKIP_genSym(SkipInt /* n */) {
```
Comment (jberdine) [inline compiler/runtime/runtime32_specific.c:96]: It isn't clear to me why this arg being unused isn't a bug. The native version of this function ensures to return a value that is larger than the arg, but the wasm one does not. @pikatchu do you know why?

**Review by jberdine — COMMENTED** [review]
Overall LGTM. There is one change noted inline that I don't know why the warning isn't pointing to a bug. It would be good to understand that before silencing the warning.

File: compiler/runtime/obstack.c (hunk 1 above)
Comment (mbouaziz) [reply compiler/runtime/obstack.c:79]: I was too lazy to check...
Looks like it's indeed clang-format: https://code.visualstudio.com/docs/cpp/cpp-ide#_code-formatting and it's using the `.clang-format` file already present.

File: compiler/runtime/runtime.c (hunk 2 above)
Comment (mbouaziz) [reply compiler/runtime/runtime.c:20]: Hmm, indeed!

File: compiler/runtime/runtime.c (hunk 3 above)
Comment (mbouaziz) [reply compiler/runtime/runtime.c:86]: There's still code in this section that seem to be related to the **global** context

File: compiler/runtime/runtime.c (hunk 2 above)
Comment (mbouaziz) [reply compiler/runtime/runtime.c:20]: Now removed

File: compiler/runtime/runtime32_specific.c (hunk 4 above)
Comment (mbouaziz) [reply compiler/runtime/runtime32_specific.c:96]: Opened https://github.com/SkipLabs/skdb/issues/141 so we can keep moving

**Review by jberdine — APPROVED** [review]

File: compiler/runtime/Makefile [hunk 5]
```diff
@@ -1,8 +1,12 @@
 MEMSIZE32=1073741824
 
+COMMONFLAGS=-Wall -Wextra -Wno-sign-conversion -Wno-sometimes-uninitialized -Wno-c2x-extensions -Wsign-compare -Wextra-semi-stmt
```
Comment (gregsexton) [inline compiler/runtime/Makefile:3]: I would probably throw in -Wconversion too. That tends to catch a lot of useful things.

**Comment by gregsexton** [comment]
This is great! I've wanted to do this for a long time :D. If you've got cycles and fancy it: I was also thinking of throwing in ubsan and asan for debug builds.

File: compiler/runtime/Makefile (hunk 5 above)
Comment (mbouaziz) [reply compiler/runtime/Makefile:3]: :+1: 

I actually did `-Weverything` and looked at warnings that appeared and could be worth fixing later.

**Comment by mbouaziz** [comment]
> I was also thinking of throwing in ubsan and asan for debug builds.

Created https://intern.skiplabs.io/tasks/3bHdAtQiSR_wcnxJtJ9uX_XdsUC for it ;-)

**Comment by gregsexton** [comment]
Do we have -werror somewhere? Or are these all just warnings?

**Comment by mbouaziz** [comment]
They are just warnings for now. werror would be good if people agree.

**Comment by bennostein** [comment]
> They are just warnings for now. werror would be good if people agree.

Bring on the `-Werror`!

**Comment by jberdine** [comment]
> > They are just warnings for now. werror would be good if people agree.
> 
> Bring on the `-Werror`!

At least for the release profile, but I have to manually set CXXFLAGS when iterating on the compiler anyhow so adding `-Wno-error` ought to work.

**Pull request merged**